test/gpu/octree/test_knn_search.cpp

```diff
@@ -114,6 +114,8 @@ TEST(PCL_OctreeGPU, exactNeighbourSearch)
             
     //prepare output buffers on device
     pcl::gpu::NeighborIndices result_device(data.tests_num, k);    
+    pcl::gpu::Octree::ResultSqrDists result_sqr_distances;
+    result_sqr_distances.create(data.tests_num * k);
```

Comment:
The constructor allows you to specify directly the number of elements you want to have.